main/lsp/LSPIndexer.cc

```diff
@@ -236,14 +236,14 @@ LSPFileUpdates LSPIndexer::commitEdit(SorbetWorkspaceEditParams &edit) {
 
     // Index changes in initialGS. pipeline::index sorts output by file id, but we need to reorder to match the order of
     // other fields.
-    UnorderedMap<u2, int> fileToPos;
```

Comment:
On the slow path, this map will contain every file in the project. Files with IDs > 2^16 will overwrite entries of files with IDs < 2^16 in this map. One of those files will be left with a `nullptr` tree, causing a crash later on.